### PR TITLE
StringUtils.java: rewrite some functions (4)

### PR DESCRIPTION
**Pull request by abderraouf-adjal**
StringUtils.java: cleanup and rewrite some functions:

- `toOtherCase()`
- `startsWithIgnoreCase()`
- `truncate()`

### CODE REVIEW

File: src/org/limewire/util/StringUtils.java [hunk 1]
```diff
@@ -179,24 +184,7 @@ private static final int subset(String little, int littleStart, int littleStop,
...
-        else if (i <= Z) //upper-case
-            return (char) (i + SHIFT);
-        else if (i < a) //non alphabetic
-            return c;
-        else if (i <= z) //lower-case
-            return (char) (i - SHIFT);
-        else
-            //non alphabetic
-            return c;
+       return Character.isLowerCase(c) ? Character.toUpperCase(c) : Character.toLowerCase(c);
```
Comment (gubatron) [inline src/org/limewire/util/StringUtils.java]: please revert this to be the old way.

**Comment by abderraouf-adjal** [comment]
@gubatron check the output, it was going to be a disaster ! http://ideone.com/0ZsoYI

**Comment by gubatron** [comment]
I see the output, I see how they differ.
But how do you know which one is correct from the two?

You mean, the old limewire version was correct, and this is why LimeWire had it like this?

**Comment by abderraouf-adjal** [comment]
@gubatron in English chars there is no problem, but what if that used with an Arabic or Russian string? it will damage the the string bytes.
So it is better to do the changes in bytes from 65 to 122 only. And BTW the original function can be with less lines by using `Character.to[Lower | toUpper]Case(c)` with bytes from 65 to 122 only.

**Pull request closed without merging**